x-pack/platform/plugins/shared/inference_endpoint/server/lib/get_inference_endpoint.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+export const getInferenceEndpoint = async (
+  esClient: ElasticsearchClient,
+  inferenceEndpointId: string
+) => {
+  try {
+    await esClient.inference.get({ inference_id: inferenceEndpointId });
+    return true;
+  } catch (err: any) {
+    if (err?.body?.error?.type === 'resource_not_found_exception') {
```

Comment:
Would it be better to check for status code instead? 
```suggestion
    if (err.statusCode === 404) {
```